AppiumForMac/Server/Handlers/AfMHandlers.m

```diff
@@ -238,7 +238,17 @@ - (AppiumMacHTTPJSONResponse *)post_url:(NSString*)path data:(NSData*)postData
 // Refresh the current page.
 
 // POST /session/:sessionId/execute
-// Inject a snippet of JavaScript into the page for execution in the context of the currently selected frame. The executed script is assumed to be synchronous and the result of evaluating the script is returned to the client.
+// execute a shell command on the mac target
+-(AppiumMacHTTPJSONResponse *)post_execute:(NSString*)path data:(NSData*)postData
+{
+    return [self executeWebDriverCommandWithPath:path data:postData onMainThread:YES commandBlock:^(AfMSessionController *session, NSDictionary *commandParams, int *statusCode)
+            {
+                NSString *script = (NSString*)[commandParams objectForKey:@"script"];
```

Comment:
It would also make sense to have a possibility to provide custom environment variables.

Reply:
Since it takes a shell script as a parameter you can specify environment variables in the shell script. And I'm not sure how useful it would be. I'd think about implementing it if someone open's an issue about it.